### DISCUSSION TITLE
What (open source) epub reader do you use to read (verify) output from Quarto?

### DISCUSSION THREAD

**Discussion by fuhrmanator** [quarto-dev/quarto-cli#4152] [Q&A]
My book project is nearing a release at the end of February. Some weeks ago, I tried to view the ePub output (it's not an essential format for me) using a couple of tools on Windows, with very mixed results. In one case, with [Aquile Reader](https://apps.microsoft.com/store/detail/aquile-reader-modern-epub-ebook-reader/9P08T4JLTQNK?hl=en-ca&gl=ca&rtc=1), two columns in a table seemed to have inverted the width specifications (vs. the HTML and PDF formats that are great). Other results were very mixed - figures (I use SVG a lot) were badly cropped or even cut off. I came to a quick conclusion after looking at those tools (on Windows) that unless I want to install Calibre (which is frankly huge), there don't seem to be many options. 

For the students who'll read my book in ePub format (again, not required), I would like to recommend something robust, smaller than Calibre, and open source. Sumatra PDF will read an epub file, but it seems it's just presented much like a PDF - not really an e-reader.

What tool(s) do you use for EPUB and why? Do you convert them to Kindle? Please mention if the tools are open source or free and on which platform (Windows, MacOS, Linux).

**Comment by RichardJActon** [comment]
- As you say Calibre is a venerable and very full featured cross platform FLOSS ebook reader https://calibre-ebook.com/
- foliate is a nice simple modern viewer for the Linux desktop https://johnfactotum.github.io/foliate/ also FLOSS
- For your windows users still looking for a FLOSS app They might want to try peruse https://apps.kde.org/en-gb/peruse/ from the KDE project see also https://peruse.kde.org/ That's a pretty lightweight ebook reader compared to calibre

**Reply by fuhrmanator** [reply]
Thanks for the answer.

I didn't know about peruse and tried installing it, but the build is from 2016 (it installs in Windows 10 with lots of error messages in the console which closes rapidly before I could diagnose). It doesn't seem to run when I launch it (nothing starts up).

I also realized that Microsoft dropped epub support (in Edge) in 2019, which might explain why there are not so many options these days (or why FLOSS projects like Aquile are stagnant?). https://mspoweruser.com/microsoft-to-end-epub-book-support-in-edge/ 

Calibre seems to be the best option so far; I wish there was a distribution of just its reader - that would be a simpler option for neophyte users of epub content.

**Comment by richardsprague** [comment]
Another free option (though not open source) is to use the ePub validator built into Apple's [ePub Upload](https://authors.apple.com/epub-upload/start#new-upload)  for Apple Book. You need an Apple ID, but otherwise the submission process is simple. If you're not worried about Apple getting a copy of your file (they say they throw it away), I find it a quick and easy way to do a sanity check on an ePub.

**Reply by fuhrmanator** [reply]
Do you have experience with this in Quarto? Is the feedback reasonable? I'm coming down off of a couple of PDF-A validation tools that just seem out of touch with reality (e.g., using SVG images instead of bitmaps causes problems, etc.).

**Reply by fuhrmanator** [reply]
So, I have an Apple ID, but it requires a little more than that apparently. The iTunes store wants me to designate that I will distribute books, to supply tax and banking information, etc. However, if I say I will just publish my book for free, I don't need to give all that. 

I must have had to do multi-factor authentication on my iPad 4 times (so many logins, at least on Windows)! 

But now it's letting me upload. I see lots of errors related to the SVG files. This is promising, provided they are correctable. Some of these SVG files are generated in the Pandoc pipeline, so we'll see.

**Comment by RichardJActon** [comment]
I cam across this new epub reader project on flathub Koodo reader, has .dmg and .exe packages for mac and windows.
https://koodo.960960.xyz/en

**Comment by fuhrmanator** [comment]
I also discovered https://www.w3.org/publishing/epubcheck/ which is a Java-based command-line tool. If I'm not mistaken, Apple's verification uses it, since the error messages are similar (at least for my Quarto project).

**Comment by MaGnus-Li** [comment]
I just read them in zotero
